packages/block-library/src/post-content/edit.js

```diff
@@ -45,40 +56,120 @@ function ReadOnlyContent( {
 	);
 }
 
-function EditableContent( { context = {} } ) {
-	const { postType, postId } = context;
+function EmptyContentPlaceholder( { context, onClose, openInserter } ) {
+	const { postType } = context;
+	const label =
+		'page' === postType
+			? __( 'This page’s content is empty' )
+			: __( 'This post’s content is empty' );
+	return (
+		<Placeholder
+			icon={ icon }
+			label={ label }
+			instructions={ __(
+				'Add your first block or pattern to get started.'
+			) }
+		>
+			<Button variant="primary" onClick={ openInserter }>
+				{ __( 'Choose a pattern' ) }
+			</Button>
+
+			<Button variant="secondary" onClick={ onClose }>
+				{ __( 'Start blank' ) }
+			</Button>
+		</Placeholder>
+	);
+}
 
+function PostContentPlaceholder( { layoutClassNames } ) {
+	const blockProps = useBlockProps( { className: layoutClassNames } );
+	return (
+		<div { ...blockProps }>
+			<Placeholder
+				className="wp-block-post-content__content-placeholder"
+				withIllustration={ true }
+			>
+				<p>
+					{ __( 'This block will be replaced with your content.' ) }
+				</p>
+			</Placeholder>
+		</div>
+	);
+}
+
+function EditableContent( { context = {}, clientId } ) {
+	const { postType, postId } = context;
+	const { selectBlock, insertBlock } = useDispatch( blockEditorStore );
 	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
 		'postType',
 		postType,
 		{ id: postId }
 	);
 
-	const entityRecord = useSelect(
+	const { entityRecord, setInserterIsOpened } = useSelect(
 		( select ) => {
-			return select( coreStore ).getEntityRecord(
-				'postType',
-				postType,
-				postId
-			);
+			return {
+				entityRecord: select( coreStore ).getEntityRecord(
+					'postType',
+					postType,
+					postId
+				),
+				setInserterIsOpened:
+					select( blockEditorStore ).getSettings()
+						.__experimentalSetIsInserterOpened,
+			};
 		},
 		[ postType, postId ]
 	);
 
 	const hasInnerBlocks = !! entityRecord?.content?.raw || blocks?.length;
+	const [ hasPlaceholder, setHasPlaceholder ] = useState( ! hasInnerBlocks );
 
-	const initialInnerBlocks = [ [ 'core/paragraph' ] ];
-
-	const props = useInnerBlocksProps(
-		useBlockProps( { className: 'entry-content' } ),
+	const { children, ...props } = useInnerBlocksProps(
+		useBlockProps( {
+			className: classnames( 'entry-content', {
+				'wp-block-post-content__placeholder': hasPlaceholder,
+			} ),
+		} ),
 		{
 			value: blocks,
 			onInput,
 			onChange,
-			template: ! hasInnerBlocks ? initialInnerBlocks : undefined,
 		}
 	);
-	return <div { ...props } />;
+
+	useEffect(
+		() => setHasPlaceholder( ! hasInnerBlocks ),
+		[ hasInnerBlocks ]
+	);
```

Comment:
Brain a little fuzzy from my extended leave but is this `useEffect` necessary? It's derived state, right? i.e. we can just use `const hasPlaceholder = ! hasInnerBlocks`.

Reply:
Well, slap my hogs, you're right. I'll remove that. Another copy paste drama.